### PR TITLE
migration: Add the case of migration_host

### PR DESCRIPTION
**Pull request by cliping**
VIRT-293806 - VM live migration - network data transport -
 TCP - migration_host

Signed-off-by: cliping <lcheng@redhat.com>

### CODE REVIEW

**Comment by cliping** [comment]
Test result:

#avocado run --vt-type libvirt --vt-machine-type q35 migration.migration_uri.network_data_transport.tcp.migration_host
No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : ccc113f19ca711859b8a88edef0e11e1b9200ef1
JOB LOG    : /var/lib/avocado/job-results/job-2022-06-27T02.34-ccc113f/job.log
 (01/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_precopy.p2p: STARTED
 (01/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_precopy.p2p: PASS (206.92 s)
 (02/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_precopy.non_p2p: STARTED
 (02/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_precopy.non_p2p: PASS (204.86 s)
 (03/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_postcopy.p2p: STARTED
 (03/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_postcopy.p2p: PASS (239.45 s)
 (04/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_postcopy.non_p2p: STARTED
 (04/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv4.with_postcopy.non_p2p: PASS (239.05 s)
 (05/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_precopy.p2p: STARTED
 (05/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_precopy.p2p: PASS (206.14 s)
 (06/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_precopy.non_p2p: STARTED
 (06/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_precopy.non_p2p: PASS (206.29 s)
 (07/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_postcopy.p2p: STARTED
 (07/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_postcopy.p2p: PASS (239.87 s)
 (08/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_postcopy.non_p2p: STARTED
 (08/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_dest_qemu_conf_ipv6.with_postcopy.non_p2p: PASS (236.62 s)
 (09/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_precopy.p2p: STARTED
 (09/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_precopy.p2p: PASS (215.72 s)
 (10/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_precopy.non_p2p: STARTED
 (10/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_precopy.non_p2p: PASS (217.91 s)
 (11/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_postcopy.p2p: STARTED
 (11/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_postcopy.p2p: PASS (249.99 s)
 (12/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_postcopy.non_p2p: STARTED
 (12/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv4.with_postcopy.non_p2p: PASS (249.42 s)
 (13/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_precopy.p2p: STARTED
 (13/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_precopy.p2p: PASS (216.21 s)
 (14/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_precopy.non_p2p: STARTED
 (14/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_precopy.non_p2p: PASS (218.39 s)
 (15/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_postcopy.p2p: STARTED
 (15/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_postcopy.p2p: PASS (250.20 s)
 (16/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_postcopy.non_p2p: STARTED
 (16/16) type_specific.io-github-autotest-libvirt.migration.migration_uri.network_data_transport.tcp.migration_host.set_src_hosts_ipv6.with_postcopy.non_p2p: PASS (249.20 s)
RESULTS    : PASS 16 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2022-06-27T02.34-ccc113f/results.html
JOB TIME   : 3653.61 s

File: libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tcp.py [hunk 1]
```diff
@@ -1,26 +1,96 @@
+from avocado.utils import process
+
 from virttest import libvirt_version
+from virttest import remote
+from virttest import libvirt_remote
+from virttest import utils_net
```
Comment (Yingshun) [inline libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tcp.py:6]: Could you please update them in alphabet order?

**Review by Yingshun — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tcp.py (hunk 1 above)
Comment (cliping) [reply libvirt/tests/src/migration/migration_uri/migration_network_data_transport_tcp.py]: Updated, thanks.

**Review by Yingshun — APPROVED** [review]

File: libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tcp.cfg [hunk 2]
```diff
@@ -23,6 +23,9 @@
     server_pwd = "${migrate_dest_pwd}"
     status_error = "no"
     check_network_accessibility_after_mig = "yes"
+    uri_port = "16509"
```
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tcp.cfg]: Update to migrate_desturi_port. Thanks.

File: libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tcp.cfg [hunk 3]
```diff
@@ -38,7 +41,23 @@
...
+            action_during_mig = '[{"func": "libvirt_network.check_established", "after_event": "iteration: '1'", "func_param": 'params'}]'
+            ipv6_addr_des = 'ENTER.YOUR.IPv6.DESTINATION'
+            server_cn = "ENTER.YOUR.EXAMPLE.SERVER_CN"
+            variants:
+                - set_dest_qemu_conf_ipv4:
+                    qemu_conf_dest = '{r".*migration_host\s*=.*": "migration_host='${migrate_dest_host}'"}'
+                - set_dest_qemu_conf_ipv6:
+                    ipv6_config = 'yes'
+                    qemu_conf_dest = '{r".*migration_host\s*=.*": "migration_host='${ipv6_addr_des}'"}'
+                - set_src_hosts_ipv4:
```
[the remark this answers is not in the document]
Comment (cliping) [reply libvirt/tests/cfg/migration/migration_uri/migration_network_data_transport_tcp.cfg]: Update to default_src_connect_ipv4/default_src_connect_ipv6.

**Review by fangge1212 — APPROVED** [review]

**Pull request merged**